### ISSUE TITLE
Strange compile dependency to wiremock 

### ISSUE THREAD

**Issue by mariuszs** [4finance/micro-infra-spring#398]
Is `wiremock` dependency required for *compile* in `boot-microservice`?

'''
gradle dependencyInsight --configuration testCompile --dependency wiremock

com.github.tomakehurst:wiremock:1.54 (conflict resolution)
\--- testCompile

com.github.tomakehurst:wiremock:1.52 -> 1.54
+--- com.ofg:micro-deps-spring-test-config:0.8.17
|    \--- com.ofg:micro-infra-spring-test:0.8.17
|         \--- testCompile
\--- com.ofg:stub-runner:0.8.17
     \--- com.ofg:stub-runner-spring:0.8.17
          \--- com.ofg:micro-deps-spring-config:0.8.17
               +--- com.ofg:micro-deps-spring-test-config:0.8.17 (*)
               \--- com.ofg:micro-infra-spring-base:0.8.17
                    \--- com.ofg:micro-infra-spring:0.8.17
                         \--- com.ofg:micro-infra-spring-boot-starter:0.8.17
'''

**Comment by marcingrzejszczak** [comment]
It's needed for the DEV mode. It's starting your stubs in Wiremock.

**Comment by mariuszs** [comment]
Ok, but this is included in production/stage too. Maybe we should limit this to dev profile only?

Btw. wiremock is in old version

**Comment by marcingrzejszczak** [comment]
Ad 1) How do you want to add it to a jar only in dev and test Spring app mode?
Ad 2) There is a PR with Wiremock upgrading.

**Comment by mariuszs** [comment]
i think we dont need to add this jar to micro-infra jar at all. What we need, is a profile in microservice like DEV with dependency to wiremock.

**Comment by marcingrzejszczak** [comment]
We've agreed that we could extract Dev mode related stuff to a separate jar and include that ONLY when running the app from Gradle task

**Comment by szpak** [comment]
Similar situation seems to be with `curator-test`:
```
[INFO] +- com.ofg:micro-infra-spring-base:jar:0.8.19:compile
[INFO] |  +- org.aspectj:aspectjweaver:jar:1.8.5:runtime
[INFO] |  +- com.ofg:micro-deps-spring-config:jar:0.8.19:compile
[INFO] |  |  +- com.ofg:micro-deps:jar:0.8.19:compile
[INFO] |  |  |  +- org.apache.curator:curator-x-discovery:jar:2.7.1:compile
[INFO] |  |  |  |  +- org.apache.curator:curator-recipes:jar:2.7.1:compile
[INFO] |  |  |  |  |  \- org.apache.curator:curator-framework:jar:2.7.1:compile
[INFO] |  |  |  |  |     \- org.apache.curator:curator-client:jar:2.7.1:compile
[INFO] |  |  |  |  \- org.apache.zookeeper:zookeeper:jar:3.4.6:compile
[INFO] |  |  |  |     +- log4j:log4j:jar:1.2.17:compile
[INFO] |  |  |  |     +- jline:jline:jar:0.9.94:compile
[INFO] |  |  |  |     \- io.netty:netty:jar:3.7.0.Final:compile
[INFO] |  |  |  +- org.apache.curator:curator-test:jar:2.7.1:compile
[INFO] |  |  |  |  +- org.javassist:javassist:jar:3.18.1-GA:compile
(...)
```

**Comment by marcingrzejszczak** [comment]
Related to the in memory Zookeeper

**Comment by szpak** [comment]
Yes, I added it to remember to move curator-test and embedded zookeeper server to that brand new package for `dev` mode.
